Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -734,3 +763,165 @@ getStats driverId date = do
+  Redis.setExp alternateNumberAttemptsKey attempts expTime
+
+validateAlternateNumber ::
+  ( EsqDBFlow m r,
+    EncFlow m r,
+    Redis.HedisFlow m r,
+    HasCacheConfig r,
+    CoreMetrics m,
+    HasFlowEnv m r ["apiRateLimitOptions" ::: APIRateLimitOptions, "smsCfg" ::: SmsConfig],
+    HasFlowEnv m r '["altenateOtpSmsTemplate" ::: Text],
+    HasAternateNumberAttempts r
+  ) =>
+  Id SP.Person ->
+  DriverAlternateNumberReq ->
+  m APISuccess
+validateAlternateNumber personId phoneNumber = do
+  Redis.whenWithLockRedis (makeAlternatePhoneNumberKey personId) 60 $ do
+    mobileNumberHash <- getDbHash phoneNumber.alternateNumber
+    person <- QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+    duplicateCheck (QPerson.findByMobileNumber phoneNumber.mobileCountryCode mobileNumberHash) "Person with this mobile number already exists"
+    smsCfg <- asks (.smsCfg)
+    let useFakeOtpM = useFakeSms smsCfg
+    let fakeOtp = show <$> useFakeOtpM
+    otpCode <- maybe generateOTPCode return fakeOtp
+    otpSmsTemplate <- asks (.altenateOtpSmsTemplate)
+    let otpHash = smsCfg.credConfig.otpHash
+    let altPhoneNumber = phoneNumber.mobileCountryCode <> phoneNumber.alternateNumber
+    let sender = smsCfg.sender
+    altNoAttempt <- asks (.alternateNumberAttempts)
+    withLogTag ("personId_" <> getId person.id) $
+      Sms.sendSMS person.merchantId (Sms.constructSendSMSReq otpCode otpHash otpSmsTemplate altPhoneNumber sender) >>= Sms.checkSmsResult
+    cacheAlternateNumberInfo personId phoneNumber.alternateNumber otpCode altNoAttempt
+  return Success
+  where
+    duplicateCheck cond err = whenM (isJust <$> cond) $ throwError $ InvalidRequest err
+
+validateAuthVerifyReq :: Validate DriverAlternateNumberOtpReq
+validateAuthVerifyReq DriverAlternateNumberOtpReq {..} =
+  sequenceA_
+    [ validateField "otp" otp $ ExactLength 4 `And` star P.digit
+    ]
+
+verifyHitsCountKey :: Id SP.Person -> Text
+verifyHitsCountKey id = "Driver:AlternateNumberOtp:verify:" <> getId id <> ":hitsCount"
+
+verifyAlternateNumberOtp ::
+  ( HasFlowEnv m r '["apiRateLimitOptions" ::: APIRateLimitOptions],
+    EsqDBFlow m r,
+    Redis.HedisFlow m r,
+    EncFlow m r,
+    CoreMetrics m,
+    HasCacheConfig r
+  ) =>
+  Id SP.Person ->
+  DriverAlternateNumberOtpReq ->
+  m APISuccess
+verifyAlternateNumberOtp personId req = do
+  Redis.whenWithLockRedis (makeAlternatePhoneNumberKey personId) 60 $ do
+    runRequestValidation validateAuthVerifyReq req
+    person <- QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+    checkSlidingWindowLimit (verifyHitsCountKey personId)
+    val <- Redis.get (makeAlternateNumberOtpKey personId)
+    authValueHash <- case val of
+      Nothing -> throwError $ InternalError "Auth not found"
+      Just a -> return a
+    unless (authValueHash == req.otp) $ throwError InvalidAuthData
+    value <- Redis.get (makeAlternatePhoneNumberKey personId)
+    altMobNo <- case value of
```

Comment:
you can use `fromMaybeM` for this case:

`altMobNo <- Redis.get (makeAlternatePhoneNumberKey personId) >>= fromMaybeM (InternalError "Alternate Number not found")`